### PR TITLE
[Fix](Github issue fixes)

### PR DESCRIPTION
**Pull request by cb-amutha**
1. Chargebee.configure never finishes on error
2. Fixed CBEnvironment iOS build issue
3. Code refactor on chargebee wrapper class

### CODE REVIEW

File: lib/src/utils/product.dart [hunk 1]
```diff
@@ -61,6 +62,7 @@ class Subscripton {
     customerId = json['customer_id'] as String;
```
Comment (cb-sriramthiagarajan) [inline lib/src/utils/product.dart:62]: Should `customerId` be added as an instance variable too?

File: lib/src/chargebee.dart [hunk 2]
```diff
@@ -10,11 +8,15 @@ import 'dart:convert';
...
   static const platform = MethodChannel(Constants.methodChannelName);
+  static String status="";
+  static List<CBItem> listItems = [];
+  static List<Product> products = [];
+  static List<Subscripton> subscriptions = [];
+  static List<CBPlan> listPlans = [];
 
 /* Configure the app details with chargebee system */
-  static Future<void> configure(String site, String publishableApiKey,
+  static Future<String> configure(String site, String publishableApiKey,
```
Comment (cb-sriramthiagarajan) [inline lib/src/chargebee.dart:18]: Can we define an enum for status?

File: lib/src/chargebee.dart [hunk 3]
```diff
@@ -29,233 +31,128 @@ class Chargebee {
...
-    String purchaseResult = await platform.invokeMethod(
-        Constants.mPurchaseProduct,
-        {Constants.product: product.id, Constants.customerId: customerId});
-    if (purchaseResult.isNotEmpty) {
-      return PurchaseResult.fromJson(jsonDecode(purchaseResult.toString()));
-    } else {
-      return PurchaseResult("", purchaseResult);
-    }
+      if (customerId == null) customerId = "";
+      else customerId = customerId;
```
Comment (cb-sriramthiagarajan) [inline lib/src/chargebee.dart:58]: Do we need this else block? 🤔

**Review by cb-sriramthiagarajan — CHANGES_REQUESTED** [review]

File: lib/src/chargebee.dart (hunk 3 above)
Comment (cb-amutha) [reply lib/src/chargebee.dart]: No, its removed.

File: lib/src/chargebee.dart (hunk 2 above)
Comment (cb-amutha) [reply lib/src/chargebee.dart]: As discussed, status been removed on configure.

File: lib/src/utils/product.dart (hunk 1 above)
Comment (cb-sriramthiagarajan) [reply lib/src/utils/product.dart:62]: Okay, this is invalid

Comment (cb-amutha) [reply lib/src/utils/product.dart:62]: `customerId` instance variable of Subscription class

File: lib/src/chargebee.dart [hunk 4]
```diff
@@ -31,231 +28,128 @@ class Chargebee {
...
+  static Future<List> retrieveProductIdentifers(
+      [Map<String, String>? queryParams]) async {
+      String result = await platform.invokeMethod(
+          Constants.mProductIdentifiers, queryParams);
+      return jsonDecode(result);
   }
 
   /* Get entitlement details from chargebee system */
-  static Future<List<dynamic>> retrieveEntitlements(
+  static Future<List> retrieveEntitlements(
```
Comment (cb-sriramthiagarajan) [inline lib/src/chargebee.dart:100]: We should also define the type of this `List`, but I think we can take it up separately.

**Review by cb-sriramthiagarajan — APPROVED** [review]

File: lib/src/chargebee.dart (hunk 4 above)
Comment (cb-amutha) [reply lib/src/chargebee.dart:100]: Yes, we can do it as part of improvements task.

**Pull request merged**